File: src/test/java/javabot/operations/ForgetFactoidOperationTest.java
package javabot.operations;

import java.io.IOException;

import javabot.BotEvent;
import javabot.dao.ChangeDao;
import javabot.dao.FactoidDao;
import org.springframework.beans.factory.annotation.Autowired;
import org.testng.Assert;
import org.testng.annotations.Test;

@Test(groups = {"operations"})
public class ForgetFactoidOperationTest extends BaseOperationTest {
    @Autowired
    private FactoidDao factoidDao;

    @Autowired
    private ChangeDao changeDao;

    public void forgetFactoid() {
        if (!factoidDao.hasFactoid("afky")) {
            factoidDao.addFactoid(SENDER, "afky", "test");
        }
        testOperation("forget afky", "I forgot about afky, " + SENDER + ".", "Should have forgotten factoid");
    }

    public void nonexistantFactoid() {
        testOperation("forget asdfghjkl", "I never knew about asdfghjkl anyway, " + SENDER + ".", "Should not have known about factoid");
    }

    @Override
    protected BotOperation createOperation() {
        return new ForgetFactoidOperation(getJavabot());
    }

    public void channelMessage() throws IOException {
        final BotEvent event = new BotEvent("##javabot", SENDER, "", "localhost", "pong is");
        Assert.assertEquals(getOperation().handleChannelMessage(event).size(), 0, "Should be an empty list");
    }
}

File: src/test/kotlin/javabot/operations/ForgetFactoidOperationTest.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import javabot.BaseTest
import javabot.dao.FactoidDao
import javabot.dao.LogsDaoTest
import javax.inject.Inject
import org.testng.Assert
import org.testng.annotations.Test

@Test
class ForgetFactoidOperationTest : BaseTest() {

    @Inject private lateinit var factoidDao: FactoidDao
    @Inject private lateinit var operation: ForgetFactoidOperation

    fun forgetFactoid() {
        if (!factoidDao.hasFactoid("afky")) {
            factoidDao.addFactoid(TEST_USER.nick, "afky", "test", LogsDaoTest.CHANNEL_NAME)
        }
        var response = operation.handleMessage(message("~forget afky"))
        Assert.assertEquals(response[0].value, Sofia.factoidForgotten("afky", TEST_USER.nick))
    }

    fun nonexistentFactoid() {
        var response = operation.handleMessage(message("~forget asdfghjkl"))
        Assert.assertEquals(
            response[0].value,
            Sofia.factoidDeleteUnknown("asdfghjkl", TEST_USER.nick),
        )
    }
}
